### PR TITLE
[PIR]Migrate graph_khop_sampler into pir

### PR DESCRIPTION
**Pull request by changeyoung98**
### PR types

Others
### PR changes

APIs
### Description

pcard-67164
Migrate graph_khop_sampler into pir

### CODE REVIEW

File: paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,426 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zyfncg) [inline paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc]: 2022 -> 2024

File: paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+  std::copy(dst_merge.begin(), dst_merge.end(), p_out_dst);
+}
+}  // namespace phi
+
+PD_REGISTER_KERNEL(graph_khop_sampler,
+                   CPU,
+                   ALL_LAYOUT,
+                   phi::GraphKhopSamplerKernel,
+                   int,
+                   int64_t) {}
```
Comment (zyfncg) [inline paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc]: sample_index需要设置下DataType，可参考：
image

File: paddle/phi/kernels/graph_khop_sampler_kernel.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zyfncg) [inline paddle/phi/kernels/graph_khop_sampler_kernel.h]: 2022 -> 2024

File: paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc (hunk 2 above)
Comment (changeyoung98) [reply paddle/phi/kernels/cpu/graph_khop_sampler_kernel.cc]: 已修复

**Review by zyfncg — DISMISSED** [review]

**Review by lanxianghit — DISMISSED** [review]

**Review by risemeup1 — DISMISSED** [review]

**Review by yuanlehome — DISMISSED** [review]

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM, no docs changes

**Review by zyfncg — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by yuanlehome — APPROVED** [review]

**Review by risemeup1 — APPROVED** [review]

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Pull request merged**